diff --git a/lib/rapidsms/backends/email.py b/lib/rapidsms/backends/email.py
index 52b280c..1abd29f 100644
--- a/lib/rapidsms/backends/email.py
+++ b/lib/rapidsms/backends/email.py
@@ -37,23 +37,16 @@ class Backend(backend.Backend):
     _title = "Email"
     _connection = None
     
-    def configure(self, host="localhost", port=25, username="demo-user@domain.com", 
-                  password="secret", use_tls=True, fail_silently=False):
+    def configure(self, host="localhost", port=25, fail_silently=False):
         # the default information will not work, users need to configure this
         # in their settings
         
         # this is some commented out code that doesn't call django email packages
-        self._username = username 
         self._host = host
         self._port = port
-        self._password = password
-        self._use_tls = use_tls 
         self._fail_silently = fail_silently
-        self._connection = SMTPConnection(username=username,
-                                          port=port,
+        self._connection = SMTPConnection(port=port,
                                           host=host,
-                                          password=password,
-                                          use_tls=use_tls,
                                           fail_silently=fail_silently)
     
     def send(self, message):
@@ -121,4 +114,4 @@ class Backend(backend.Backend):
         return (subject, from_string, to_ple, email_message.get_payload())
      
     
-        
\ No newline at end of file
+        

